jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -574,7 +589,7 @@ public void checkpointJVM() {
 		/* Add env variables restore hook */
 		registerRestoreEnvVariables();
 
-		if (isCRIUSupportEnabled()) {
+		if (isCheckpointAllowed()) {
 			checkpointJVMImpl(imageDir, leaveRunning, shellJob, extUnixSupport, logLevel, logFile, fileLocks, workDir,
 					tcpEstablished, autoDedup, trackMemory);
 		} else {
```

Comment:
if its not allowed we need to check if it is enabled so we can output the correct message. Either:
1) It is not enabled
2) It is enabled but its running in non-portable mode so only one checkpoint is allowed it we have already checkpointed once

Reply:
Throwing `UnsupportedOperationException` within `isCheckpointAllowed()`.

Reply:
sorry I didnt mean throw the exception in `isCheckpointAllowed`, but I meant when we throw in `checkpointJVM` we should check if if it is enabled so we know what message to output

Reply:
sure, will address this along w/ https://github.com/eclipse-openj9/openj9/issues/14676 since the api in question will be moved.

Reply:
Throwing `UnsupportedOperationException` w/ messages according to `isCRIUSupportEnabled`.
Also added changes to address https://github.com/eclipse-openj9/openj9/issues/14676

@tajila please have another look.